stdlib/public/core/Sequence.swift

```diff
@@ -583,6 +583,18 @@ extension SequenceType where Generator.Element : Equatable {
     return split(maxSplit, allowEmptySlices: allowEmptySlices,
       isSeparator: { $0 == separator })
   }
+
+  /// Return an `Array` containing the elements of `self`
+  @warn_unused_result
+  public func uniq() -> [Generator.Element] {
+    return reduce(Array<Generator.Element>()) { (arr, element) -> [Generator.Element] in
```

Comment:
@rothomp3 beat me to it :) yes reduce can really suffer performance-wise. here's an elegant answer I stumbled upon a few month back

```swift
func uniq<S: SequenceType, E: Hashable where E == S.Generator.Element>(source: S) -> [E] {
  var seen = [E: Bool]()
  return source.filter { seen.updateValue(true, forKey: $0) == nil }
}
```

http://stackoverflow.com/a/33984316/870155